Bulk message fetch, deletion, and example clear command.

ZeroSumGame opened the pull request:
This adds in the ability to get messages in bulk from the server.  I've also added in an example "clear" command, which is one of the most common features found in Discord bots.

b1naryth1ef commented on src/dscord/api/client.d:
```diff
@@ -240,10 +275,27 @@ class APIClient {
+    filter with respect to supplied messageID.
+  */
+  Message[] channelsMessagesList(Snowflake chan, uint limit = 50, MessageFilter filter = MessageFilter.BEFORE, Snowflake msg = 0){
+    assert(limit <= 100, "The maximum number of messages that can be deleted at one time is 100.");
```
This assert message seems wrong. Should also likely be a fully runtime check, assert gets compiled out in release which seems iffy for required validation stuff like this (e.g. we know the request will fail, might as well not make the HTTP round trip to Discord API servers)

b1naryth1ef commented on examples/src/basic.d:
```diff
@@ -43,6 +43,36 @@ class BasicPlugin : Plugin {
+
+    if(messages.length > 0){
+      this.client.deleteMessages(event.msg.channelID, messages);
+      auto msg = event.msg.replyf("I deleted %s messages for you.", messages.length);
```
```d
event.msg.replyf("I deleted %s messages for you.", messages.length).after(3.seconds).del();
```

ZeroSumGame commented:
@b1naryth1ef  Any word on this?  It's been almost a week.

b1naryth1ef commented:
Ah sorry I forgot to hit the submit review button.

b1naryth1ef commented on examples/src/basic.d:
```diff
@@ -43,6 +43,36 @@ class BasicPlugin : Plugin {
 
+  //An example command that clears messages in the channel
+  @Command("clear")
+  void onClearMessages(CommandEvent event){
```
Nit; If you could make the bracket spacing line up with everything else (e.g. `) {` that would be awesome.

b1naryth1ef commented on examples/src/basic.d:
```diff
@@ -43,6 +43,36 @@ class BasicPlugin : Plugin {
+    uint limit = 100;
+
+    //This command can take an integer argument.
+    if(event.args[].length > 0){
```
Why `event.args[]` and not `event.args`?

b1naryth1ef commented on examples/src/basic.d:
```diff
@@ -43,6 +43,36 @@ class BasicPlugin : Plugin {
+    //This command can take an integer argument.
+    if(event.args[].length > 0){
+      try {
+        limit = std.conv.to!int(event.args[0]);
```
preferred syntax here is just `event.args[0].to!int`

ZeroSumGame commented:
I believe this addresses all the issues.  There are lots of other places that this project could use some exception handling and build-time checks, TBH.

Any idea what is causing TravisCI to be upset with the builds?  Everything is working fine on the newest DMD compiler, so the build script may need updating.
